front/src/modules/trainschedule/components/TimetableV2/TimetableToolbar.tsx

```diff
@@ -59,7 +58,7 @@ const TimetableToolbar = ({
 
   const [deleteTrainSchedules] = enhancedEditoastApi.endpoints.deleteV2TrainSchedule.useMutation();
 
-  useTrainSchedulesDetails(
+  const uniqueTags = useTrainSchedulesDetails(
```

Comment:
why get rid of the useMemo?

Reply:
If you test the component on dev, when you select one of the tags, the other non selected are disappearing. This is because this useMemo was based on the state trainScheduleDetails and not the "pure" list of timetable trains. So when we were selecting a tag filter, trainScheduleDetails state was also filtered with the selected tag only and that made the other tags disappearing.
That's why I needed to set the uniqueTags directly from the trains fetched and not from the state :)

Reply:
Ok thank you !!